### ISSUE TITLE
CPD metadata file, spaces between values of the 4x4 matrix

### ISSUE THREAD

**Issue by spiegelss** [PDAL/PDAL#2355]
Hello,

I am currently having an issue with the CPD algorithm.  The metadata file that comes from the Coherent Point Drift algorithm does not put spaces between all of the values of the 4x4 matrix. Here's an example of the 4x4 string from the metadata.json file:

"  0.993673   0.109721  0.0239994    2.96117 -0.110056   0.993839  0.0131346    186.727-0.0224104 -0.0156927   0.999626    3.49454         0          0          0          1"

I would like a space between 186.727-0.0224104 if possible.

For this the Docker version of PDAL is being used.  

Attached is a csv and pipeline.

Thank you.

[Fixed.txt](https://github.com/PDAL/PDAL/files/2814275/Fixed.txt)
[cpd_pipeline.txt](https://github.com/PDAL/PDAL/files/2814278/cpd_pipeline.txt)

**Comment by gadomski** [comment]
As I don't have `Moving.las`, I can't run your example pipeline. I did check the CPD result using your `Fixed.txt` for both the fixed and the moving matrices, and got the following transform, which has no whitespace problems:

```           
1  7.11237e-17 -1.38778e-17  1.66925e-14  2.1001e-16            1  2.77556e-17  3.76642e-14-1.21431e-17  1.38778e-17            1  9.12898e-16           0            0            0            1
```

Can you do two things?

1. Check to make sure there isn't a unprinted newline between the 186.727 and the -0.0224104.
2. Post your Eigen version here. As the result matrix is just an [Eigen::MatrixXd](https://github.com/gadomski/cpd/blob/master/include/cpd/matrix.hpp#L29), the printed representation of the result matrix is defined in Eigen, not by PDAL's CPD plugin.

**Comment by spiegelss** [comment]
Here is the moving las file as text.  
[Moving.txt](https://github.com/PDAL/PDAL/files/2814600/Moving.txt)

link for the docker file:
https://hub.docker.com/r/pdal/pdal/dockerfile

Thank you again.

**Comment by abellgithub** [comment]
I believe this is happening because we are removing control characters from JSON strings.  The separator for rows in Eigen, by default, is a newline.  I'm not sure why we're removing control characters rather than escaping them.  Looking at this history, this seems like a copy/paste from Stackoverflow, which in this case is wrong. 

https://github.com/PDAL/PDAL/blob/master/pdal/util/Utils.cpp#L424-L433

Thoughts @connormanning, @hobu?

**Comment by gadomski** [comment]
I'd prefer whitespace over an escaped `\n`, but an escaped `\n` is better than nothing at all.

**Comment by abellgithub** [comment]
Fixed in #2525
